tests/TestCase/View/Helper/PaginatorHelperTest.php

```diff
@@ -1533,6 +1533,226 @@ public function testNumbers()
+            9 => '1 2 ... 5 6 7 8 *9 10 11 12 13 ',
+            10 => '1 2 ... 6 7 8 9 *10 11 12 13 14 ',
+            15 => '1 2 ... 11 12 13 14 *15 16 17 18 19 ',
+            16 => '1 2 ... 12 13 14 15 *16 17 18 19 20 ',
+            17 => '1 2 ... 12 13 14 15 16 *17 18 19 20 ',
+            18 => '1 2 ... 12 13 14 15 16 17 *18 19 20 ',
+            19 => '1 2 ... 12 13 14 15 16 17 18 *19 20 ',
+            20 => '1 2 ... 12 13 14 15 16 17 18 19 *20 ',
+        ];
+        $result = $this->getNumbersForMultiplePages(array_keys($expected), 20, ['first' => 2]);
+        $this->assertEquals($expected, $result);
+
+        $expected = [
+            1 => '*1 2 3 4 5 6 7 8 9 ... L> ',
+            2 => '1 *2 3 4 5 6 7 8 9 ... L> ',
+            3 => '1 2 *3 4 5 6 7 8 9 ... L> ',
+            4 => '1 2 3 *4 5 6 7 8 9 ... L> ',
+            5 => '1 2 3 4 *5 6 7 8 9 ... L> ',
+            6 => '2 3 4 5 *6 7 8 9 10 ... L> ',
+            7 => '3 4 5 6 *7 8 9 10 11 ... L> ',
+            8 => '4 5 6 7 *8 9 10 11 12 ... L> ',
+            9 => '5 6 7 8 *9 10 11 12 13 ... L> ',
+            10 => '6 7 8 9 *10 11 12 13 14 ... L> ',
+            11 => '7 8 9 10 *11 12 13 14 15 ... L> ',
+            12 => '8 9 10 11 *12 13 14 15 16 ... L> ',
+            13 => '9 10 11 12 *13 14 15 16 17 ... L> ',
+            14 => '10 11 12 13 *14 15 16 17 18 19 20 ',
+            15 => '11 12 13 14 *15 16 17 18 19 20 ',
+            16 => '12 13 14 15 *16 17 18 19 20 ',
+            17 => '12 13 14 15 16 *17 18 19 20 ',
+            18 => '12 13 14 15 16 17 *18 19 20 ',
+            19 => '12 13 14 15 16 17 18 *19 20 ',
+            20 => '12 13 14 15 16 17 18 19 *20 ',
+        ];
+        $result = $this->getNumbersForMultiplePages(array_keys($expected), 20, ['last' => 'L']);
+        $this->assertEquals($expected, $result);
+
+        $expected = [
+            1 => '*1 2 3 4 5 6 7 8 9 ... L> ',
+            2 => '1 *2 3 4 5 6 7 8 9 ... L> ',
+            3 => '1 2 *3 4 5 6 7 8 9 ... L> ',
+            4 => '1 2 3 *4 5 6 7 8 9 ... L> ',
+            5 => '1 2 3 4 *5 6 7 8 9 ... L> ',
+            6 => '1 2 3 4 5 *6 7 8 9 10 ... L> ',
+            7 => '1 2 3 4 5 6 *7 8 9 10 11 ... L> ',
+            8 => '<F ... 4 5 6 7 *8 9 10 11 12 ... L> ',
+            9 => '<F ... 5 6 7 8 *9 10 11 12 13 ... L> ',
+            10 => '<F ... 6 7 8 9 *10 11 12 13 14 ... L> ',
+            11 => '<F ... 7 8 9 10 *11 12 13 14 15 ... L> ',
+            12 => '<F ... 8 9 10 11 *12 13 14 15 16 ... L> ',
+            13 => '<F ... 9 10 11 12 *13 14 15 16 17 ... L> ',
+            14 => '<F ... 10 11 12 13 *14 15 16 17 18 19 20 ',
+            15 => '<F ... 11 12 13 14 *15 16 17 18 19 20 ',
+            16 => '<F ... 12 13 14 15 *16 17 18 19 20 ',
+            17 => '<F ... 12 13 14 15 16 *17 18 19 20 ',
+            18 => '<F ... 12 13 14 15 16 17 *18 19 20 ',
+            19 => '<F ... 12 13 14 15 16 17 18 *19 20 ',
+            20 => '<F ... 12 13 14 15 16 17 18 19 *20 ',
```

Comment:
Agree but normally such data sets are candidates for PHPUnit's DataProvider:

https://phpunit.de/manual/5.7/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers